src/test/java/org/broadinstitute/hellbender/engine/ReadsDataSourceUnitTest.java

```diff
@@ -49,17 +49,26 @@ public void testHandleNonExistentFile() {
     @Test(expectedExceptions = UserException.class)
     public void testHandleUnindexedFileWithIntervals() {
         // Cannot initialize a reads source with intervals unless all files are indexed
-        ReadsDataSource readsSource = new ReadsDataSource(new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam"));
+        final File unindexed = new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam");
+        assertUnindexed(unindexed);
+        ReadsDataSource readsSource = new ReadsDataSource(unindexed);
         readsSource.setIntervalsForTraversal(Arrays.asList(new SimpleInterval("1", 1, 5)));
     }
 
     @Test(expectedExceptions = UserException.class)
     public void testHandleUnindexedFileQuery() {
         // Construction should succeed, since we don't pass in any intervals, but the query should throw.
-        ReadsDataSource readsSource = new ReadsDataSource(new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam"));
+        final File unindexed = new File(READS_DATA_SOURCE_TEST_DIRECTORY + "unindexed.bam");
+        assertUnindexed(unindexed);
```

Comment:
Change this to `Assert.assertNotNull(SamFiles.findIndex(unindexed), "Expected file to have no index, but found an index file");`